DotBot.py

```diff
@@ -96,6 +105,12 @@ def receive(self, frame):
         self.simEngine.schedule(self.nextBumpTime, self._bumpSensorCb, tag=f'{self.dotBotId}_bumpSensorCb')
         log.debug(f'next bump for {self.dotBotId} scheduled for {self.nextBumpTime}')
 
+    def getDeviceData(self):
+        '''
+        for wireless PDR computations and caching
+        '''
+        return {'deviceId': self.dotBotId, 'position': (self.x, self.y)}
```

Comment:
deviceId!=dotBotId !!!!!!!!!!!!!!!!!!!!!!!!!!

Reply:
to match the function signature at the orchestrator, sometimes the device is the orchestrator. I cant assure to get sender.getDeviceData()['dotBotId'] is the sender is the orchestrator and wireless isn't supposed to differentiate between orchestrator and dotbot